v2/internal/attacktechniques/gcp/exfiltration/share-compute-image/main.go

```diff
@@ -0,0 +1,182 @@
+
+import (
+	compute "cloud.google.com/go/compute/apiv1"
+	"context"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"log"
+
+	_ "embed"
+	"fmt"
+	"github.com/datadog/stratus-red-team/v2/internal/providers"
+	gcp_utils "github.com/datadog/stratus-red-team/v2/internal/utils/gcp"
+	"github.com/datadog/stratus-red-team/v2/pkg/stratus"
+	"github.com/datadog/stratus-red-team/v2/pkg/stratus/mitreattack"
+)
+
+//go:embed main.tf
+var tf []byte
+
+const codeBlock = "```"
+const AttackTechniqueId = "gcp.exfiltration.share-compute-image"
+
+func init() {
+	stratus.GetRegistry().RegisterAttackTechnique(&stratus.AttackTechnique{
+		ID:           "gcp.exfiltration.share-compute-image",
+		FriendlyName: "Exfiltrate Compute Image by sharing it",
+		IsSlow:       true,
+		Description: `
+Exfiltrates a Compute Image by sharing with a fictitious attacker account. The attacker could then create a snapshot of the image in their GCP project.
+
+Warm-up:
+
+- Create a Compute Image
+
+Detonation:
+
+- Set the IAM policy of the image so that the attacker account has permissions to read the image in their own project
+
+!!! note
+
+	Since the target e-mail must exist for this attack simulation to work, Stratus Red Team grants the role to ` + gcp_utils.DefaultFictitiousAttackerEmail + ` by default.
+	This is a real Google account, owned by Stratus Red Team maintainers and that is not used for any other purpose than this attack simulation. However, you can override
+	this behavior by setting the environment variable <code>` + gcp_utils.AttackerEmailEnvVarKey + `</code>, for instance:
+
+	` + codeBlock + `bash
+	export ` + gcp_utils.AttackerEmailEnvVarKey + `="your-own-gmail-account@gmail.com"
+	stratus detonate ` + AttackTechniqueId + `
+	` + codeBlock + `
+`,
+		Detection: `
+You can detect when someone changes the IAM policy of a Compute Image, using the GCP Admin Activity audit logs event <code>v1.compute.images.setIamPolicy</code>. Here's a sample event, shortened for clarity:
+
+` + codeBlock + `json hl_lines="18 20 25""
+{
+  "protoPayload": {
+    "@type": "type.googleapis.com/google.cloud.audit.AuditLog",
+    "authenticationInfo": {
+      "principalEmail": "user-sharing-the-image@domain.tld",
+      "principalSubject": "user:user-sharing-the-image@domain.tld"
+    },
+    "requestMetadata": {
+      "callerIp": "34.33.32.31",
+      "callerSuppliedUserAgent": "google-cloud-sdk gcloud/..."
+    },
+    "resourceName": "projects/victim-project/global/images/stratus-red-team-victim-image",
+    "request": {
+      "policy": {
+        "version": "3",
+        "bindings": [
+          {
+            "role": "roles/owner",
+            "members": [
+              "user:attacker@gmail.com"
+            ]
+          }
+        ]
+      },
+      "@type": "type.googleapis.com/compute.images.setIamPolicy"
+    }
+  }
+}
+` + codeBlock + `
+
+After the attacker has permissions on the Compute Image, they can export it in their own GCP Storage using:
```

Comment:
does the attacker need permissions on the image before doing this? wouldn't it be simpler to not share it, and directly export it to an attacker-controlled bucket?

Reply:
From what I understood, sharing the image relies on the fact that the attacker obtain permissions to use `compute.images.setIamPolicy`. After doing so, the attacker can freely access the image from her own project.

On the other hand, exporting to an attacker-controlled bucket is much more involved as it requires using the Cloud Build API, and therefore require to either have permissions to enable it, or permissions to use it. Also, it is likely to be much noisier. 

Nevertheless, the latter would make an interesting use-case to add to `stratus-red-team` too.